src/confluent_kafka/aio/producer/_message_batch.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2025 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import namedtuple
+from typing import List, Any, Optional
+
+
+# Create immutable MessageBatch value object
+MessageBatch = namedtuple('MessageBatch', [
+    'topic',        # str: Target topic for this batch
+    'messages',     # tuple: Prepared message dictionaries with callbacks
+    'futures',      # tuple: asyncio.Future objects to resolve
+    'callbacks'     # tuple: User callback functions (can be None)
+])
+
+
+def create_message_batch(topic: str, messages: List[dict], futures: List[Any], callbacks: List[Optional[callable]]) -> MessageBatch:
+    """Create an immutable MessageBatch from lists
+    
+    This factory function converts mutable lists into an immutable MessageBatch object.
+    
+    Args:
+        topic: Target topic name
+        messages: List of prepared message dictionaries
+        futures: List of asyncio.Future objects
+        callbacks: List of user callback functions
+        
+    Returns:
+        MessageBatch: Immutable batch object
+    """
+    return MessageBatch(
+        topic=topic,
+        messages=tuple(messages),
```

Comment:
Is the only reason to copy the list / tuple here? Do we need to ensure an immutable list passed? If so you could do a `messages = tuple(messages) if not isinstance(messages, tuple) else messages` to avoid copy if already immutable.